packages/block-library/src/embed/edit.js

```diff
@@ -192,6 +192,26 @@ const EmbedEdit = ( props ) => {
 		web: sprintf( __( '%s URL' ), title ),
 		native: title,
 	} );
+
+	const onSubmit = ( event ) => {
+		if ( event ) {
+			event.preventDefault();
+		}
+
+		setIsEditingURL( false );
+		setAttributes( { url } );
+	};
+
+	const onSubmitNative = ( value ) => {
+		// On native, the URL change is only notified when submitting
+		// so we have to explicitly set the URL.
+		setURL( value );
+
+		// Replicate the same behavior as onSubmit
+		setIsEditingURL( false );
+		setAttributes( { url: value } );
+	};
```

Comment:
I tried to replicate the web flow for dispatching URL updates and submitting but I couldn't find a reliable way to achieve it. The approach I tried to follow was using the `onChange` callback when the `setAttributes` callback of `LinkSettingsNavigation ` component is triggered. The problem is that `setAttributes` can be called two times if you tap on the submit/intro button on the keyboard, in fact, the second call is happening after dismissing the bottom sheet so it updates the URL after calling `onSubmit` which is not expected.

Here you can watch a debugging session of the described issue:

https://user-images.githubusercontent.com/14905380/123791721-10bc9380-d8e0-11eb-83d5-b827d4d0d6c9.mp4

For this reason, I had to create a native version of the `onSubmit` callback, which instead of an event it receives the URL value and does both updating the URL and submitting.

Reply:
Want to update the inline comment (`On native, the URL change is only notified when submitting`) to include something like `, and not via 'onChange'`? I had to debug the code to verify that the [`url` state](https://github.com/WordPress/gutenberg/blob/b5ee74ef9c968b194221e0d4e0d97dc04b479fd3/packages/block-library/src/embed/edit.js#L65) is indeed not getting updated from [this onChange](https://github.com/WordPress/gutenberg/blob/b5ee74ef9c968b194221e0d4e0d97dc04b479fd3/packages/block-library/src/embed/edit.js#L230), which is a bit unintuitive. 😬

Reply:
Good idea, I'll update it.

I always found a bit confusing the case of having a list of props but not knowing which one are only for web version and others for native. In this case, the [onChange](https://github.com/WordPress/gutenberg/blob/b5ee74ef9c968b194221e0d4e0d97dc04b479fd3/packages/block-library/src/embed/edit.js#L230) prop is only used in the web version.

Reply:
let's do it in a new PR? The CI tests just got all-green after multiple re-runs 😬

Reply:
Ok, let's do it that way, looks like we're lucky that all CI tests are 🟢 😅 .